dotnet/src/SemanticKernel/SkillDefinition/SKFunction.cs

```diff
@@ -554,9 +554,8 @@ private static MethodDetails GetMethodDetails(MethodInfo methodSignature, object
         else if (skMainParam != null)
         {
             // The developer used [SKFunctionInput] on a function that doesn't support a string input
```

Comment:
(preexisting problem) I don't believe this exception is valid -- if the user doesn't specify a string, they can still get the "input" from SKContext. This should only be thrown if the function has neither a SKContext arg nor a string arg.

Reply:
There could be functions having no parameters like those from TimeSkill class. Is it something we plan to keep as it's or do we plan to update them to accept SKContext even so they don't need it?

Reply:
supporting functions without a context allows some flexibility which I believe is nice to have

Reply:
@shawncal `SKFunctionInputAttribute` is used for `string input`. It's true that one can pass the input via the context, but the `SKFunctionInputAttribute` attribute is specific to the string param